spec/models/order_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'rails_helper'
+
+RSpec.describe Order, type: :model do
+  describe "Associations" do
+    it{is_expected.to belong_to(:user)}
+    it{is_expected.to have_many(:order_details).dependent(:destroy)}
+    it{is_expected.to have_many(:books).through(:order_details)}
+  end
+
+  describe "Enums" do
+    it {
+      should define_enum_for(:status)
+        .with_values(pending: 0, rejected: 1, accepted: 2,
+                     shipped: 3, delivered: 4, canceled: 5)
+        .with_prefix
+    }
+  end
+
+  describe "Validations" do
+    before {FactoryBot.build :order}
+
+    context "with delivery address field" do
+      it{is_expected.to validate_presence_of(:delivery_address)}
+    end
+
+    context "with delivery phone field" do
+      it{is_expected.to validate_presence_of(:delivery_phone)}
+      it{is_expected.to validate_length_of(:delivery_phone).is_at_most(Settings.digit_10)}
+    end
+  end
+
+  describe "private methods" do
+    let!(:order){FactoryBot.create :order}
```

Comment:
phân biệt let vs let!

Reply:
@anlvq-1058 `let! `thì kết quả được tính toán và cache lại trước khi khối it chạy
còn `let` chỉ khi nào nó thực sự được gọi, nó mới được tạo và nó sẽ được cache lại cho đến khi test kết thúc ạ